File: 75-sort-colors/sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let start = 0;
    let next = 1;
    let end = nums.length - 1;
    let color = 0;
    while (color < 3) {
        if (next > end) {
            //continue from previous start position and loop for new color
            next = start+1;
            color++;
        }
        if (nums[start] == color) {
            start++;
            next++;
        } else if (nums[start] != nums[next] && nums[next] == color) {
            nums[next] = nums[start]
            nums[start] = color;
            start++;
        } else {
            next++;
        }
    }
    return nums;
};

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        start = 0
        end = len(nums) - 1
        iteration = 0
        while iteration<=end:
            if nums[iteration] == 0:
                nums[iteration] = nums[start]
                nums[start] = 0
                start +=1
                iteration +=1
            elif nums[iteration] == 2:
                nums[iteration] = nums[end]
                nums[end] = 2
                end -=1
            else:
                iteration +=1
        